aips/staking-contract-to-delegation-pool.md

```diff
@@ -0,0 +1,107 @@
+
+## Impact
+The resulting delegation pool proxies the existing stake pool of the staking contract.
+
+The operator is a config of the underlying stake pool, and thus it remains unchanged.
+
+The commission fee from the staking contract is preserved. However, changing the commission fee will apply from the next lockup cycle from now on.
+
+The pool ownership is preserved, the staker becomes owner of the delegation pool and can set the operator and commission fee as before.
+
+Partial voting is automatically enabled. Previously, the staker could set the delegated voter of the underlying stake pool, implicitly of their owned stake. Now, the staker can vote and delegate their voting power as a regular delegator.
+
+If the beneficiary of operator is not configured on `aptos_framework::delegation_pool`, then it cannot be initialized as the conversion function doesn't have access to operator's signer, only to staker's. The default beneficiary will be used which is the operator themselves.
+If the beneficiary is already set, then this address will be used for the new delegation pool as well.
+
+In terms of rewards, there is virtually no impact on the staker, they continue to be rewarded at the same rate as before. In case of the operator, they may expect higher rewards as additional delegators join the pool.
+
+Staker's delegated voter (stake pool's voter) is preserved and applies immediately. However, delegating voting power will apply from the next lockup cycle from now on.
+
+To summarize, state that will be preserved:
+- pool ownership
+- operator
+- commission fee
+- delegated voter of staker
+- individual stakes of staker, operator and any previous operators still owning pending-inactive commission
+
+While, state that will change:
+- the voter of the stake pool is set to the resource account of the delegation pool
+- beneficiary of operator: address already applying on delegation-pool module will be used
+
+## Alternative solutions
+
+As described above, manually converting to a delegation pool requires:
+- doubling the infrastructure for a brief period
+- rewards downtime
+- decreased network security for a brief period
+
+## Specification
+
+`aptos_framework::delegation_pool` implements `initialize_delegation_pool_from_staking_contract(staker: &signer, operator: address)` which is called by `staker` on their staking contract identified by `operator` to atomically convert it, preserving the state of the underlying stake pool, implicitly the activeness of the validator.
+
+`aptos_framework::staking_contract` implements `destroy_staking_contract(staker: &signer, operator: address)` which will only be called by `aptos_framework::delegation_pool::initialize_delegation_pool_from_staking_contract`. This function destroys the `StakingContract` between `staker` and `operator` and returns the resource account storing the stake pool and the ownership capability.
```

Comment:
Could you list all the resources to be deleted? (e.g., `BeneficiaryForOperator`?)